File: PruebaBBDD/PruebaBBDD/Form1.vb
Imports MySql.Data.MySqlClient
Public Class Form1
    Dim conector As New MySqlConnection("server=localhost;user=ejemplo;password=ejemplo;database=ejemplo;port=3306")
    Dim comando As String
    Dim reader As MySqlDataReader

    Function Validar(ByVal numDep As Integer) As Boolean
        comando = "SELECT * FROM Departamentos WHERE dept_no = @num"
        Dim cmd As New MySqlCommand(comando, conector)
        conector.Open()
        cmd.Parameters.Add("@num", MySqlDbType.Int16).Value = numDep
        reader = cmd.ExecuteReader
        Validar = False

        If reader.Read Then
            reader.Close()
            Validar = True
        End If

        conector.Close()
        Return Validar
    End Function


    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

    End Sub

    Private Sub Consultar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Consultar.Click
        comando = "SELECT dnombre, loc FROM Departamentos WHERE dept_no = @numDep"
        Dim cmd As New MySqlCommand(comando, conector)
        conector.Open()
        cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = TextBox1.Text
        reader = cmd.ExecuteReader

        If reader.Read Then
            TextBox2.Text = reader.GetString(0)
            TextBox3.Text = reader.GetString(1)
        Else
            MsgBox("El departamento no existe")
        End If
        reader.Close()
        conector.Close()

    End Sub

    Private Sub Añadir_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Añadir.Click
        'comando = "SELECT dept_no FROM Departamentos WHERE dept_no = @numDep"
        'Dim cmd As New MySqlCommand(comando, conector)

        'cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = TextBox1.Text
        'reader = cmd.ExecuteReader

        If Validar(TextBox1.Text) Then
            MsgBox("El departamento ya existe")
        Else
            reader.Close()
            Dim comando As String = "INSERT INTO Departamentos VALUES(@numDep, @nom, @loc)"
            Dim cmd As New MySqlCommand(comando, conector)
            conector.Open()
            cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = TextBox1.Text
            cmd.Parameters.Add("@nom", MySqlDbType.String).Value = TextBox2.Text
            cmd.Parameters.Add("@loc", MySqlDbType.String).Value = TextBox3.Text
            cmd.ExecuteNonQuery()
            MsgBox("Departamento Introducido")
        End If


        conector.Close()
    End Sub

    Private Sub Eliminar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Eliminar.Click
        comando = "SELECT dept_no FROM Departamentos WHERE dept_no = @numDep"
        Dim cmd As New MySqlCommand(comando, conector)
        conector.Open()
        cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = TextBox1.Text
        reader = cmd.ExecuteReader

        If reader.Read Then
            Try
                reader.Close()
                Dim comando2 As String = "DELETE FROM Departamentos WHERE dept_no = @numDep"
                Dim cmd2 As New MySqlCommand(comando2, conector)
                cmd2.Parameters.Add("@numDep", MySqlDbType.Int16).Value = TextBox1.Text
                cmd2.ExecuteNonQuery()

                MsgBox("Departamentos Eliminado")
            Catch ex As MySqlException
                MsgBox(ex.Message)
            End Try
           


        Else
            MsgBox("El departamento no existe")
        End If

        ' reader.Close()
        conector.Close()

    End Sub

    Private Sub Modificar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Modificar.Click
        comando = "UPDATE Departamentos SET dnombre = @nom, loc = @loc WHERE dept_no = @numDep"
        Dim cmd As New MySqlCommand(comando, conector)
        conector.Open()
        cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = TextBox1.Text
        cmd.Parameters.Add("@nom", MySqlDbType.String).Value = TextBox2.Text
        cmd.Parameters.Add("@loc", MySqlDbType.String).Value = TextBox3.Text

        cmd.ExecuteNonQuery()
        MsgBox("El Departamento ha sido modificado")

        conector.Close()
    End Sub
End Class

File: BBDDC/BBDDC/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

namespace WindowsFormsApplication1{
    public partial class Form1 : Form{
        MySqlConnection conector = new MySqlConnection("server=localhost;user=ejemplo;password=ejemplo;database=ejemplo");
        String comando;
        MySqlCommand cmd;
        MySqlDataReader reader;

        public Form1(){            
            InitializeComponent();
        }

        public Boolean Validar(Int16 num)
        {
            Boolean existe = false;

            comando = "SELECT * FROM Departamentos WHERE dept_no = @num";
            cmd = new MySqlCommand(comando, conector);
            conector.Open();
            cmd.Parameters.Add("@num", MySqlDbType.Int16).Value = num;
            reader = cmd.ExecuteReader();

            if (reader.Read())
            {
                reader.Close();
                existe = true;
            }

            conector.Close();
            return existe;
        }

        public void Limpiar()
        {
            textBox1.Text = "";
            textBox2.Text = "";
            textBox3.Text = "";
        }

        private void Consultar_Click(object sender, EventArgs e){
            comando = "SELECT * FROM Departamentos WHERE dept_no = @numDep";
            cmd = new MySqlCommand(comando, conector);
            conector.Open();
            cmd.Parameters.Add("@numDep", MySqlDbType.Byte).Value = textBox1.Text;
            reader = cmd.ExecuteReader();

            if (reader.Read())
            {
                textBox2.Text = reader.GetString(1);
                textBox3.Text = reader.GetString(2);
            }
            else
            {
                MessageBox.Show("El departamento no existe", "Error");
            }
            conector.Close();
        }

        private void Añadir_Click(object sender, EventArgs e)
        {
            

            if (!Validar(Convert.ToInt16(textBox1.Text)))
            {
                comando = "INSERT INTO Departamentos VALUES(@numDep, @nom, @loc)";
                cmd = new MySqlCommand(comando, conector);
                conector.Open();
                cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = textBox1.Text;
                cmd.Parameters.Add("@nom", MySqlDbType.String).Value = textBox2.Text;
                cmd.Parameters.Add("@loc", MySqlDbType.String).Value = textBox3.Text;

                cmd.ExecuteNonQuery();
                MessageBox.Show("Departamento introducido", "Exito");
                Limpiar();
            }
            else
            {
                MessageBox.Show("El departamento ya existe", "Error");
            }
            conector.Close();
        }

        private void Eliminar_Click(object sender, EventArgs e)
        {
            if (Validar(Convert.ToInt16(textBox1.Text)))
            {
                try
                {
                    comando = "DELETE FROM Departamentos WHERE dept_no = @numDep";
                    cmd = new MySqlCommand(comando, conector);
                    conector.Open();
                    cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = textBox1.Text;

                    cmd.ExecuteNonQuery();

                    MessageBox.Show("Departamento eliminado", "Exito");
                    Limpiar();
                }
                catch (MySqlException me)
                {
                    MessageBox.Show(me.Message, "");
                }
            }
            else {
                MessageBox.Show("El departamento no existe", "Error");
            }

            conector.Close();
        }

        private void Modificar_Click(object sender, EventArgs e)
        {
            if (Validar(Convert.ToInt16(textBox1.Text)))
            {
                comando = "UPDATE Departamentos SET dnombre = @nom, loc = @loc WHERE dept_no = @numDep";
                cmd = new MySqlCommand(comando, conector);
                conector.Open();
                cmd.Parameters.Add("@numDep", MySqlDbType.Int16).Value = textBox1.Text;
                cmd.Parameters.Add("@nom", MySqlDbType.String).Value = textBox2.Text;
                cmd.Parameters.Add("@loc", MySqlDbType.String).Value = textBox3.Text;

                cmd.ExecuteNonQuery();
                MessageBox.Show("Departamento modificado", "Exito");
                Limpiar();
            }
            else
            {
                MessageBox.Show("El departamento no existe");            
            }
            conector.Close();
        }    
    }
}
